pkg/client/userd/grpc.go

```diff
@@ -52,6 +55,18 @@ func (s *service) logCall(c context.Context, callName string, f func(context.Con
 }
 
 func (s *service) withSession(c context.Context, callName string, f func(context.Context, trafficmgr.Session) error) (err error) {
+	defer func() {
```

Comment:
Not sure if this is that useful. It affects the subset of the grpc calls that *require* a session. It does not affect calls like `Connect`, `Disconnect`, `Login`, or `Quit`. It does however affect SystemA's call to `WatchWorkloads`. A side-effect is that whenever a new call is added, the implementor must remember to add an entry in the `TELEMETRY.md` (I guess the names implicitly added here should go in the too?).

In general, I think we should avoid generic approaches like this and instead focus on what we know is useful such as the outcome of intercept calls or connect attempts. The scout calls are not lightweight. They do affect performance on our side and as the number of users grow, they will consume a significant amount of resources from the metriton service.

Reply:
ok makes sense, i'll instrument the specific RPCs that we're interested in which are `CanIntercept` and `CreateIntercept`